Analytics service - server side

crookedneighbor opened the pull request:
Mixpanel removed, amplitude implemented.

I'm a little concerned that we might be tracking things both on the server and on the browser.

crookedneighbor committed:
Remove mixpanel mock

crookedneighbor committed:
Add initial server side analytics service

crookedneighbor committed:
Set up purchase tracking

crookedneighbor committed:
Create analytics.track

crookedneighbor committed:
Add analytics to utils

crookedneighbor committed:
Track cron with analytics service

crookedneighbor committed:
Add format user data function to analytics server

crookedneighbor committed:
Send only analytics for use in cron

crookedneighbor committed:
Track purchases with analytics service

crookedneighbor committed:
Track unsubscription

crookedneighbor committed:
Remove double instantiation

crookedneighbor committed:
Update how info is sent to google

crookedneighbor committed:
Impliment analytics in user actions; remove mixpanel

crookedneighbor committed:
Update amplitude version

crookedneighbor committed:
Swap out ga for analytis service in auth controller

crookedneighbor committed:
Remove gogle analytics from export

crookedneighbor committed:
Added additional analytics info

crookedneighbor committed:
Correct indentation

crookedneighbor commented:
@SabreCat

crookedneighbor committed:
Remove erroneous var

SabreCat commented on website/src/analytics.js:
```diff
@@ -0,0 +1,124 @@
+
+function _formatDataForAmplitude(data) {
+  var PROPERTIES_TO_SCRUB = ['uuid', 'user', 'purchaseValue', 'gaLabel'];
+  var event_properties = _.omit(data, PROPERTIES_TO_SCRUB);
```
Ah, this is nice. We could do something like this on the client side to remove some of the GA noise from Amplitude events.

SabreCat commented on common/script/index.coffee:
```diff
@@ -1692,8 +1809,16 @@ api.wrap = (user, main=true) ->
+
+      analyticsData = {
+        category: 'behavior',
+        gaLabel: user.flags.cronCount,
```
A GA label is usually a string (sort of a subcategory); they have a "value" field for numeric subdivisions, if we'd like to track this there.

crookedneighbor replied:
I think I interpreted the value param as value in dollars, but I guess that's not what it is for. (That's how we use it for purchases, I believe)

I don't think my track code has a way to track value. Thinking of switching to the params syntax instead of arguments: https://github.com/peaksandpies/universal-analytics#event-tracking

SabreCat replied:
Values might double as dollars when doing purchase tracking, but they're a generic number field elsewhere.

SabreCat committed:
feat(analytics): Server tweaks

Remove duplicate browser-side tracking for new user registration. Move quest tracking to server side and expand to include all quest accept/reject actions.

SabreCat committed:
fix(analytics): Correct quest key

SabreCat committed:
fix(analytics): Include GA label for quest actions

crookedneighbor committed:
Merge pull request #2 from SabreCat/crookedneighbor/analytics-service-server-side

feat(analytics): Server tweaks

crookedneighbor committed:
Add server_side tests to gulp

crookedneighbor committed:
Add platform, change itemName to itemKey

crookedneighbor committed:
Adjust how quest drops are sent

crookedneighbor committed:
Refactor to use object method of sending google data

crookedneighbor committed:
Allow optional value to be passed in

crookedneighbor committed:
Clarify cron

crookedneighbor committed:
Automatically sends item name when item is purchased

crookedneighbor committed:
Bump version number to resolve error message

crookedneighbor committed:
Merge branch 'develop' into analytics-service-server-side

crookedneighbor committed:
Update test for new quest name

crookedneighbor commented:
New and improved! Now the analytics code will send the english version of the item name acquired _if_ it can find it.

crookedneighbor commented on common/script/index.coffee:
```diff
@@ -931,9 +1005,18 @@ api.wrap = (user, main=true) ->
+          goldCost: item.goldValue,
+          acquireMethod: 'Gold',
+          category: 'behavior'
+        }
```
@SabreCat Does this data look right to you? I added it in after merging in your changes on develop.

SabreCat replied:
Yup, looks right!

crookedneighbor committed:
Merge branch 'develop' into analytics-service-server-side
